tasks/ebpf.py

```diff
@@ -0,0 +1,125 @@
+
+
+def colored_diff(val1, val2):
+    if val1 <= val2:
+        return colored(val1 - val2, "green")
+    return colored(val1 - val2, "red")
+
+
+def stdout_or_file(filename=None):
+    @contextlib.contextmanager
+    def stdout():
+        yield sys.stdout
+
+    return open(filename, 'w') if filename else stdout()
+
+
+@task(
+    help={
+        "skip_object_files": "Do not build ebpf object files",
+        "base": "JSON file holding verifier statistics to compare against",
+        "jsonfmt": "Output in json format rather than tabulating",
+        "out": "Output file to write results to. By default results are written to stdout",
+    }
+)
+def print_verification_stats(ctx, skip_object_files=False, base=None, jsonfmt=False, out=None, debug_build=False):
+    sudo = "sudo" if not is_root() else ""
+    if not skip_object_files:
+        build_object_files(ctx)
+
+    # generate programs.go
+    use_debug_build = "USE_DEBUG_BUILDS='true' " if debug_build else ""
+    ctx.run(f"{use_debug_build}go generate pkg/ebpf/verifier/stats.go")
+
+    ctx.run("cd pkg/ebpf/verifier && go generate")
+    ctx.run("go build -tags linux_bpf pkg/ebpf/verifier/calculator/main.go")
+
+    debug = "--debug" if debug_build else ""
+    res = ctx.run(f"{sudo} ./main --directory pkg/ebpf/bytecode/build/co-re {debug}", hide=True)
+    print(res.stderr)
+    if res.exited == 0:
+        verifier_stats = json.loads(res.stdout)
+    else:
+        return
+
+    if base is None:
+        with stdout_or_file(out) as f:
+            if jsonfmt:
+                print(json.dumps(verifier_stats, indent=4), file=f)
+            else:
+                print(tabulate_stats(verifier_stats), file=f)
+
+        return
+
+    with open(base, 'r') as f:
+        base_verifier_stats = json.load(f)
+
+    stats_diff = dict()
+    for key, value in verifier_stats.items():
+        stat = dict()
+        if key not in base_verifier_stats:
+            stats_diff[key] = value
+            continue
+
+        base_value = base_verifier_stats[key]
+        stat["stack_usage"] = colored_diff(value["stack_usage"], base_value["stack_usage"])
+        stat["instruction_processed"] = colored_diff(
+            value["instruction_processed"], base_value["instruction_processed"]
+        )
+        stat["limit"] = colored_diff(value["limit"], base_value["limit"])
+        stat["verification_time"] = colored_diff(value["verification_time"], base_value["verification_time"])
+        stat["max_states_per_insn"] = colored_diff(value["max_states_per_insn"], base_value["max_states_per_insn"])
+        stat["total_states"] = colored_diff(value["total_states"], base_value["total_states"])
+        stat["peak_states"] = colored_diff(value["peak_states"], base_value["peak_states"])
+
+        stats_diff[key] = stat
+
+    with stdout_or_file(out) as f:
+        if jsonfmt:
+            print(json.dumps(stats_diff, indent=4), file=f)
+        else:
+            print(tabulate_stats(stats_diff), file=f)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Duplicated code, maybe it could be a small utility function, wdyt?